core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -466,6 +470,12 @@ public ManagedChannelImpl shutdown() {
     if (!shutdown.compareAndSet(false, true)) {
       return this;
     }
+    channelExecutor.executeLater(new Runnable() {
+      @Override
+      public void run() {
+        channelStateManager.gotoState(SHUTDOWN);
```

Comment:
Why isn't this in the `cancelIdleTimer()` Runnable later? It seems it should either be calling `drain()` or the code should just be moved.